pkg/services/cloudmigration/cloudmigrationimpl/snapshot_mgmt.go

```diff
@@ -369,3 +372,39 @@ func (s *Service) updateSnapshotWithRetries(ctx context.Context, cmd cloudmigrat
 	}
 	return nil
 }
+
+// sortFolders implements a sort such that parent folders always come before their children
+// Implementation inspired by ChatGPT, OpenAI's language model.
+func sortFolders(input []folder.CreateFolderCommand) []folder.CreateFolderCommand {
+	// Map from UID to the corresponding folder for quick lookup
+	folderMap := make(map[string]folder.CreateFolderCommand)
+	for _, folder := range input {
+		folderMap[folder.UID] = folder
+	}
+
+	// Function to get the depth of a folder based on its parent hierarchy
+	var getDepth func(uid string) int
+	getDepth = func(uid string) int {
+		if uid == "" {
+			return 0
+		}
+		folder, exists := folderMap[uid]
+		if !exists || folder.ParentUID == "" {
+			return 1
+		}
```

Comment:
should we add an `else` statement here to return if exists== true? in that case will reduce considerable the amount of recursion?